x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -182,6 +183,7 @@ export const createLifecycleExecutor =
 
         return state.trackedAlerts[alertId].alertUuid;
       },
+      ruleDataClient,
```

Comment:
I understand why we are adding the ruleDataClient here but I wonder if that's the right usage for the lifecycle alert services. 
That being said, it's either injected here, or through the InfraLib when creating the executor, but that would require a more complex wiring.

Reply:
I agree with this. Would you consider moving `fetchAlertbyAlertUUID` into `x-pack/plugins/rule_registry/server/utils/` similar to `fetchExistingAlerts` and then providing a pass through for that function instead of providing the ruleDataClient directly?

Reply:
@kdelemme @ymao1 will do 👍